src/main/app/pay/payment.ts

```diff
@@ -1,36 +1,43 @@
-export class PaymentState {
-  status: string
-  finished: boolean
+import 'reflect-metadata'
+import { Expose, Type } from 'class-transformer'
+import { MoneyConverter } from 'fees/moneyConverter'
+
+class Link {
+  readonly href: string
+  readonly method: string
+}
+
+class Links {
+  @Expose({ name: 'next_url' })
+  @Type(() => Link)
+  readonly nextUrl: Link
+
+  @Type(() => Link)
+  readonly self: Link
 
-  deserialize (input?: any): PaymentState {
-    if (input) {
-      this.status = input.status
-      this.finished = input.finished
-    }
-    return this
-  }
 }
 
 export class Payment {
-  id: string
-  amount: number
+
   reference: string
-  description: string
-  date_created: number // tslint:disable-line variable-name allow snake_case
-  state: PaymentState
 
-  static fromObject (input?: any): Payment {
-    return new Payment().deserialize(input)
-  }
+  @Expose({ name: 'date_created' })
+  dateCreated: number
+
+  status: string
+
+  amount: number
+
+  @Expose({ name: '_links' })
+  @Type(() => Links)
+  links: Links
 
   deserialize (input?: any): Payment {
     if (input) {
-      this.id = input.id
-      this.amount = input.amount
       this.reference = input.reference
-      this.description = input.description
-      this.date_created = input.date_created
-      this.state = new PaymentState().deserialize(input.state)
+      this.dateCreated = input.dateCreated
```

Comment:
That is not right, JSON contains `date_created` not `dateCreated`. Why not to use `'class-transformer'` here as well.